src/Mercury.ts

```diff
@@ -133,38 +31,31 @@ export interface IMercuryConnectOptions {
 	 *
 	 * This function will also be called in case of a reconnect
 	 */
-	onConnect?: TOnConnectHandler
+	onConnect?: OnConnectHandler
 
 	/**
 	 * Callback function to execute when Mercury has disconnected.
 	 *
 	 * This callback usually doesn't need to be implemented.
 	 * Mercury will handle cleanup of callback functions created in onConnect
 	 */
-	onDisconnect?: TOnConnectHandler
+	onDisconnect?: OnConnectHandler
 }
 
-/** The scope of the data in a subscription */
-export enum MercurySubscriptionScope {
-	/** Anonymous (eventName only): Receives this event across all organizations and locations */
-	AnonymousGlobal = 'anonymousGlobal',
-	/** Anonymous (eventName only): Receives this event across an entire organization including all locations in the organization. */
-	AnonymousOrganization = 'anonymousOrganization',
-	/** Anonymous (eventName only): Receives this event across all organizations and locations */
-	AnonymousLocation = 'anonymousLocation',
-	/** Anonymous (eventName only): Receives this event for a user */
-	AnonymousUser = 'anonymousUser',
-	/** Receives this event across all organizations and locations */
-	Global = 'global',
-	/** Receives this event across an entire organization including all locations in the organization. */
-	Organization = 'organization',
-	/** Receives this event across all organizations and locations */
-	Location = 'location',
-	/** Receives this event for a user */
-	User = 'user'
+interface IEventHandlers<
+	EventContract extends IMercuryEventContract = IMercuryEventContract,
+	EventName extends keyof EventContract = any,
+	EventSpace extends EventContract[EventName] = any
+> {
+	[eventName: string]: {
+		// TODO: Can these be strongly typed?
```

Comment:
these already look strongly typed. maybe the `= any` is causing it to go so wide the typing is not helpful.

Try 

```js
EventContract extends IMercuryEventContract = IMercuryEventContract,
EventName extends keyof EventContract = keyof EventContract 
EventSpace extends EventContract[EventName] = EventContract[EventName]

```